File: src/App/TwitterListener.js
Bridge.assembly("Widgetoko", function ($asm, globals) {
    "use strict";

    var Twitter = require("twitter");

    Bridge.define("Widgetoko.Twitter.TwitterListener", {
        fields: {
            _stream: null,
            _filter: null,
            _client: null
        },
        events: {
            OnReceived: null,
            OnError: null
        },
        ctors: {
            ctor: function (credentials, filter) {
                this.$initialize();
                this._filter = filter;

                this._client = new Twitter({ consumer_key: credentials.ApiKey, consumer_secret: credentials.ApiSecret, access_token_key: credentials.AccessToken, access_token_secret: credentials.AccessTokenSecret });
            }
        },
        methods: {
            Start: function () {
                this._stream = this._client.stream("statuses/filter", { track: this._filter });

                this._stream.on("data", Bridge.fn.bind(this, function (tweet) {
                        !Bridge.staticEquals(this.OnReceived, null) ? this.OnReceived(this, tweet) : null;
                    }));

                this._stream.on("error", Bridge.fn.bind(this, function (error) {
                        !Bridge.staticEquals(this.OnError, null) ? this.OnError(this, error) : null;
                    }));
            },
            Stop: function () {
                if (this._stream != null) {
                    this._stream.destroy();
                    this._stream = null;
                }
            }
        }
    });
});

File: src/Twitter/TwitterListener.cs
using System;
using Widgetoko.Twitter.API;

namespace Widgetoko.Twitter
{
    public class TwitterListener
    {
        private TwitterStream _stream;
        private readonly string _filter;
        private readonly API.Twitter _client;

        public TwitterListener(TwitterCredentials credentials, string filter)
        {
            _filter = filter;

            _client = new API.Twitter(new TwitterConfig
            {
                consumer_key = credentials.ApiKey,
                consumer_secret = credentials.ApiSecret,
                access_token_key = credentials.AccessToken,
                access_token_secret = credentials.AccessTokenSecret
            });
        }

        public void Start()
        {
            _stream = _client.stream("statuses/filter", new TwitterStreamConfig { track = _filter });

            _stream.onData(tweet =>
            {
                OnReceived?.Invoke(this, tweet);
            });

            _stream.onError(error =>
            {
                OnError?.Invoke(this, error);
            });
        }

        public void Stop()
        {
            if (_stream != null)
            {
                _stream.destroy();
                _stream = null;
            }
        }

        public event EventHandler<Tweet> OnReceived;
        public event EventHandler<string> OnError;
    }
}